### PR TITLE
Dataset init

### PR DESCRIPTION
**Pull request by ExtinctWolf83**
`Dataset init` has been refactored
`dataset_config.py` file has been created
`context.py` file has been refactored to consider `dataset.cfg`  file

### CODE REVIEW

File: deploifai/server/list.py [hunk 1]
```diff
@@ -42,14 +42,17 @@ def list_server(context: DeploifaiContextObj, project: str):
...
 
     elif "id" in context.local_config["PROJECT"]:
         project_id = context.local_config["PROJECT"]["id"]
         # query for workspace name from api
         fragment = """
                         fragment project on Project {
                             name
+                            account{
+                                username
+                            }
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/server/list.py:55]: Yes
Have reverted changes

File: deploifai/utilities/dataset_config.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+config dictionary structure:
+{
+    "DATA_STORAGE": {"id": string}
+}
+"""
+
+config_file_path = pathlib.Path().joinpath("dataset.cfg")
+
+config_sections = ["DATA_STORAGE"]
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/utilities/dataset_config.py:16]: okay

File: deploifai/utilities/dataset_config.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    Exception when a data directory has been initialised through Deploifai in an ML project.
+    """
+
+    def __init__(self, message):
+        super(DeploifaiDataAlreadyInitialisedError, self).__init__(message)
+
+
+def create_config_files():
+    """
+    Creates the folder .deploifai that stores all the config files.
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/utilities/dataset_config.py:30]: Yes
Refactored

File: dataset.cfg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[DATASET]
```
Comment (98sean98) [inline dataset.cfg:1]: should delete this file, and write `dataset.cfg` into `.gitignore`

**Review by 98sean98 — CHANGES_REQUESTED** [review]
Just remove the `dataset.cfg` and write it to `.gitignore`

File: deploifai/utilities/config/dataset_config.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    config = configparser.ConfigParser()
+
+    with config_file_path.open("w") as config_file:
+        config.write(config_file)
+
+
+def read_config_file() -> configparser.ConfigParser:
+    """
+    Read the config file in the existing .dataset.cfg file
```
Comment (98sean98) [inline deploifai/utilities/config/dataset_config.py:53]: `Read the `dataset.cfg` file

File: dataset.cfg (hunk 4 above)
Comment (ExtinctWolf83) [reply dataset.cfg]: added

File: deploifai/utilities/config/dataset_config.py (hunk 5 above)
Comment (ExtinctWolf83) [reply deploifai/utilities/config/dataset_config.py]: refactored

File: deploifai/utilities/config/dataset_config.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def create_config_files():
+    """
+    Creates the .dataset config files.
+    :return: None
+    """
+    global config_file_path
+    dataset_config_dir = pathlib.Path()
+    dataset_config_dir.mkdir(exist_ok=True)
```
Comment (98sean98) [inline deploifai/utilities/config/dataset_config.py:39]: This actually doesn't seem right to me. The dataset config directory should be the current working directory. Seems like `pathlib.Path()` would do that, but it doesn't make sense to do a `mkdir` here.

**Review by 98sean98 — CHANGES_REQUESTED** [review]
Just a small problem in `dataset_config`.

File: deploifai/utilities/config/dataset_config.py (hunk 6 above)
Comment (ExtinctWolf83) [reply deploifai/utilities/config/dataset_config.py:39]: yes
i was also slightly confused about this before
but this was for making .deploifai directory in local config
should have removed it before

**Review by 98sean98 — APPROVED** [review]
LGTM

**Pull request merged**